### ISSUE TITLE
Terminal becomes extremely laggy after printing large amount of CJK characters

### ISSUE THREAD

**Issue by fireattack** [microsoft/terminal#17414]
### Windows Terminal version

1.20.11381.0

### Windows build number

10.0.19045.4412

### Steps to reproduce

```python 
s = '打印一个非常长的中文句子，啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊'
for _ in range(100):
    print(s)
```

```js
var s = '打印一个非常长的中文句子，啊啊啊啊啊啊啊啊啊啊啊啊啊啊啊'
for (var i = 0; i < 100; i++) {    console.log(s)    }
```

Run any of the above script (`python` or `node`) in a WT tab. 

### Expected Behavior

It prints all the lines very quickly and you can scroll the window with ease.

### Actual Behavior

It becomes very laggy (takes seconds to print the whole thing) and after printing, the whole window becomes very laggy, you can barely scroll back and forth.

-------------

So, this problem is tricky because I can't find a reliable way to reproduce it.
All I can confirm is that:

The issue only started to happen in recent months.

It usually starts to happen after I use my computer for extended time (I leave my computer on 24/7). And once it started to appear, it will affect all the Terminal windows, old or new. And it does not matter which type, can reproduce in CMD, PS, PS7 etc. 

Furthermore, once the bug appears, the only way I can fix it is by restarting my computer. 

It only happens when printing large amount CJK characters (Chinese, Japanese, etc.). Printing ASCII characters are fine.

**Comment by crazyGru** [comment]
1. Close all instances of Windows Terminal.
2. Delete the settings file located at `%LOCALAPPDATA%\Packages\Microsoft.WindowsTerminal_8wekyb3d8bbwe\LocalState\settings.json`.
3. Restart Windows Terminal, which will generate a new settings file with default settings.

**Comment by lhecker** [comment]
You may be running out of GPU memory. The next time this happens can you check your GPU memory usage in Task Manager?

**Comment by DHowett** [comment]
Can you confirm whether you are using the "new text rendering engine", or whether you have it disabled? Version 1.20 will be the last version with the old text rendering engine, so it will be important to know if the issue is with the new one.

**Comment by fireattack** [comment]
Yes, I'm using the new "AtlasEngine". 

As for GPU memory, I think it's unlikely since I have a 16GB RTX 4060Ti, but I can check it next time this happened. 

The full [settings.json](https://github.com/user-attachments/files/15812527/settings.json) is also attached but nothing unusual.

**Comment by carlos-zamora** [comment]
@fireattack Any updates? Has this occurred again over the past few weeks /?

**Comment by fireattack** [comment]
@carlos-zamora no, i didn't see it happen lately (granted, i didn't use it heavily either).

**Comment by carlos-zamora** [comment]
Thanks! We're going to close this bug, but happy to reopen it if you see it again. Just let us know 😊

**Comment by fireattack** [comment]
This is happening again. Version: 1.21.2911.0

https://github.com/user-attachments/assets/663ee07f-9640-4c20-a6f6-5f3e4debeff5



Rendering setting:

![Image](https://github.com/user-attachments/assets/09ee8681-0138-41b8-bf13-36969a4b1a75)

GPU memory is fine:
![Image](https://github.com/user-attachments/assets/3df77687-af8a-4f67-9f4c-2de45668afb1)

**Comment by lhecker** [comment]
But you still can't reliably reproduce the issue, right? It only happens sporadically?
Do you happen to have a G-Sync or FreeSync monitor?

**Comment by fireattack** [comment]
That's right. Once it happens, it usually stays that way until I restart my computer. 

No, I don't have these computers. I do have dual-monitors.

**Comment by lhecker** [comment]
I'm not sure if I'll be able to help you with the issue. It's one of these cases where I'd need to get your hardware into my hands so that I can install [Superluminal](https://superluminal.eu/) on it and get a timeline trace right when it happens.

That aside, please make sure that you update all your drivers, in particular the Nvidia driver (your version has a severe security vulnerability). Try to change the "Graphics API" setting to Direct3D and enable the "Disable partial swap chain invalidation" setting below it.

If you want to investigate this issue further, would you be willing to compile this project yourself from one of my custom branches and run some tests once it happens again?

**Comment by fireattack** [comment]
Sure, feel free to link your custom branch so I can try it if it happens again (together with all your other suggestions.)

Also I have updated my driver to newest since the last post (and the bug then hasn't happened), but thanks for the reminder.

**Comment by fireattack** [comment]
>That aside, please make sure that you update all your drivers, in particular the Nvidia driver (your version has a severe security vulnerability). Try to change the "Graphics API" setting to Direct3D and enable the "Disable partial swap chain invalidation" setting below it.

Last time this happened, I tried to toggle everything in Rendering, no difference.

---------------

I also have a somewhat related case to this issue.

I remote desktop to my home computer frequently.

I noticed that, sometimes, when I started using my computer in-person after some remote sessions, the whole UI on Windows desktop becomes extremely laggy (dragging windows, switch windows, you name it) to a point it feels like <20fps.

To fix this, I have to turn off all Windows Terminal windows I left open (I usually have 2 to 3 open all the time), and this issue is immediately gone.

I feel like this must be related to this issue.

For what it's worth, I have 2 monitors (1 2k 75hz, 1 4k 60hz).
